src/app/users/users.component.ts

```diff
@@ -220,8 +222,21 @@ export class UsersComponent implements OnInit, OnDestroy, AfterViewInit {
     event.stopPropagation();
   }
 
+  deleteUserFromTeams(user) {
+    // //REPLACE THIS!
+    // this.couchService.findAll('teams').then(teamsDb => {

```

Comment:
1. Return an open observable to be used in the deleteUser function
2. You can use a selector with findAll so you don't have to filter
3. Use a switchMap to run the bulkDocs for deleting
```
return this.couchService.findAll('teams', { selector: { userId: user._id }).pipe(switchMap(teams => {
  return this.couchService.bulkDocs...
})
```